src/backend/storage/buffer/bufmgr.c

```diff
@@ -1201,6 +1204,13 @@ BufferAlloc(SMgrRelation smgr, char relpersistence, ForkNumber forkNum,
 
 		Assert(BUF_STATE_GET_REFCOUNT(buf_state) == 0);
 
+		if (buf->buf_id == wal_redo_buffer)
```

Comment:
We use the local buffer manager in the WAL redo process, so this needs to be in localbuf.c

Reply:
Shame on me:(
Yet another attempt.